docs/developer-guide/writing-actions-reducers.md

```diff
@@ -0,0 +1,203 @@
+
+**Note:** Remember to put all the actions .js files in the web/client/actions folder or in js/actions if you are working with custom plugins
+
+## Reducers
+
+Again quoting redux [documentation](https://redux.js.org/basics/reducers) they are:
+> Reducers specify how the application's state changes in response to actions sent to the store.
+
+Reducers are pure function that takes the previous state and an action and returns a new state
+> (previousState, action) => newState
+
+let's see an example:
+```
+ // @mapstore is an alias for dir_name/web/client (see webpack.config.js)
+ import {PAN_TO} from '@mapstore/actions/map';
+
+export function map( (state, action) => {
+        switch(action.type) {
+            case PAN_TO: {
+                return {
+                    ...state,
+                    center
+                };
+            }
+            default return state;
+        }
+    }
+);
+```
+As you can see we are changing the center of the map that triggers the panning action of the mapping library
+
+And that's it we have wrote an action and a reducers that make the map panning around.
+
+**Note:** Remember to put all the reducers .js files in the web/client/reducers folder or in js/reducers if you are working with custom plugins
+
+## Advanced usage and tips
+Sometimes you need to change a value of an item which is store inside an array or in a nested object.
+
+Let's imagine we have this object in the store:
+```
+layer: {
+    features: [object_1, object_2, ...object_n]
+}
+```
+
+And we have created an action that holds the id of the object to change and some properties
+```
+export const UPDATE_LAYER_FEATURE = 'LAYER:UPDATE_LAYER_FEATURE'
+export const updateFeature = (id, props = {}) => ({type: UPDATE_LAYER_FEATURE, id, props})
+```
+
+Then in the reducer we can have different implementations
+```
+import {UPDATE_LAYER_FEATURE} from '@mapstore/actions/layer';
+
+export function layer( (state, action) => {
+        switch(action.type) {
+            case UPDATE_LAYER_FEATURE: {
+                const features = [...state.features];
+                return {
+                    ...state,
+                    features: features.reduce((p, c) => {
+                        return c.id === action.id ? {...c, ...action.props} : c
+                    })
+                };
+            }
+            default return state;
+        }
+    }
+);
+```
+Which is good but maybe less readable than using other ways.
+
+We suggest to use lodash/fp which allows to avoid mutation of the objects (remember reducers are pure function)
+We have prepared some utilities in @mapstore/utils/ImmutableUtils like set
+
+The previous example with set() becomes:
+```
+import {findIndex} from 'lodash';
+...
+    case UPDATE_LAYER_FEATURE: {
+        const indexFeatureChanged = findIndex(state.features, id => action.id)
+        const features = indexFeatureChanged !== -1 ? set([indexFeatureChanged], action.props, state.features) : state.features;
```

Comment:
for this there is `arrayUpdate` : 
Suggest the best function: 
```
case UPDATE_LAYER_FEATURE: {
    // lodash get
    const feature = get(state.features, {id: action.id});
    // merge the old and the new feature object, replacing the existing element in the array
    return arrayUpdate("features", {...feature, action.props}, {id: action.id}, state);
}
```

Reply:
going to keep this and remove the other example